docs/hyperopt.md

```diff
@@ -1,153 +1,109 @@
 # Hyperopt
 This page explains how to tune your strategy by finding the optimal
-parameters with Hyperopt.
+parameters, process called hyperparameter optimization. The bot uses several 
+algorithms included in `scikit-optimize` package to accomplish this. The
+search will burn all your CPU cores, make your laptop sound like a fighter jet
+and still take a long time.
 
 ## Table of Contents
 - [Prepare your Hyperopt](#prepare-hyperopt)
-    - [1. Configure your Guards and Triggers](#1-configure-your-guards-and-triggers)
-    - [2. Update the hyperopt config file](#2-update-the-hyperopt-config-file)
-- [Advanced Hyperopt notions](#advanced-notions)
-    - [Understand the Guards and Triggers](#understand-the-guards-and-triggers)
+- [Configure your Guards and Triggers](#configure-your-guards-and-triggers)
+- [Solving a Mystery](#solving-a-mystery)
+- [Adding New Indicators](#adding-new-indicators)
 - [Execute Hyperopt](#execute-hyperopt)
 - [Understand the hyperopts result](#understand-the-backtesting-result)
 
-## Prepare Hyperopt
-Before we start digging in Hyperopt, we recommend you to take a look at 
-your strategy file located into [user_data/strategies/](https://github.com/freqtrade/freqtrade/blob/develop/user_data/strategies/test_strategy.py)
+## Prepare Hyperopting
+We recommend you start by taking a look at `hyperopt.py` file located in [freqtrade/optimize](https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/hyperopt.py)
  
-### 1. Configure your Guards and Triggers
+### Configure your Guards and Triggers
 There are two places you need to change in your strategy file to add a 
 new buy strategy for testing:
-- Inside [populate_buy_trend()](https://github.com/freqtrade/freqtrade/blob/develop/user_data/strategies/test_strategy.py#L278-L294).
-- Inside [hyperopt_space()](https://github.com/freqtrade/freqtrade/blob/develop/user_data/strategies/test_strategy.py#L244-L297) known as `SPACE`.
+- Inside [populate_buy_trend()](https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/hyperopt.py#L278-L294).
+- Inside [hyperopt_space()](https://github.com/freqtrade/freqtrade/blob/develop/freqtrade/optimize/hyperopt.py#L218-L229) 
+and the associated methods `indicator_space`, `roi_space`, `stoploss_space`.
 
-There you have two different type of indicators: 1. `guards` and 2. 
-`triggers`.
+There you have two different type of indicators: 1. `guards` and 2. `triggers`.
 1. Guards are conditions like "never buy if ADX < 10", or never buy if 
 current price is over EMA10.
 2. Triggers are ones that actually trigger buy in specific moment, like 
 "buy when EMA5 crosses over EMA10" or buy when close price touches lower 
```

Comment:
for both points ... can we add quotes to the 2nd example to make reading/understanding this easier?

Reply:
Uh, yes, definitely.